commit 8f065a26d49eabebc3995b8f6966b15ef7553b3e
Author: Luca Giuzzi <luca.giuzzi@gmail.com>
Date:   Tue Oct 4 13:35:06 2011 -0400

    rpc.idmapd: Dies with 'I/O possible'
    
    We have had problems on some of our machines (all Fedora 14), where
    rpc.idmapd used to die with an `I/O possible' message at (basically)
    random times. A strace suggested the issue being in nfsopen() where a
    signal type is reset before notification is disabled; a signal at just
    the right time might be the cause of the problem; see
    https://bugzilla.redhat.com/show_bug.cgi?id=684308
    
    Reviewed-by: Jeff Layton <jlayton@redhat.com>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/utils/idmapd/idmapd.c b/utils/idmapd/idmapd.c
index 19d9114..e80efb4 100644
--- a/utils/idmapd/idmapd.c
+++ b/utils/idmapd/idmapd.c
@@ -778,8 +778,8 @@ nfsopen(struct idmap_client *ic)
 	} else {
 		event_set(&ic->ic_event, ic->ic_fd, EV_READ, nfscb, ic);
 		event_add(&ic->ic_event, NULL);
-		fcntl(ic->ic_dirfd, F_SETSIG, 0);
 		fcntl(ic->ic_dirfd, F_NOTIFY, 0);
+		fcntl(ic->ic_dirfd, F_SETSIG, 0);
 		if (verbose > 0)
 			xlog_warn("Opened %s", ic->ic_path);
 	}
